fendermint/testing/materializer/src/manifest.rs

```diff
@@ -0,0 +1,223 @@
+/// if we are deploying to Calibration, or it might be a chain we provision
+/// with CometBFT and Fendermint.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum Rootnet {
+    /// Existing L1 running outside our control.
+    ///
+    /// This implies using some sort of Faucet to get balances for the accounts.
+    External {
+        /// Indicate whether we have to (re)deploy the IPC contract or we can use an existing one.
+        deployment: IpcDeployment,
+    },
+
+    /// Provision a new chain to run the L1.
+    ///
+    /// It is assumed that a newly provisioned chain will have built-in support for IPC,
+    /// e.g. the way Fendermint deploys IPC actors at well-known addresses.
+    New {
+        /// Collateral of the initial validator set.
+        validators: CollateralMap,
+        /// Balances of the accounts in the rootnet.
+        ///
+        /// These balances will go in the genesis file.
+        balances: BalanceMap,
+        /// Nodes that participate in running the root chain.
+        nodes: NodeMap,
+    },
+}
+
+/// An IPC subnet.
+///
+/// The balance of the account on the parent subnet, as declared in this manifest,
+/// _does not_ have to account for the collateral/balance we have to take from it to join/fund the subnet.
+/// When we create the testnet configuration from the manifest we will account for this with a rollup,
+/// so we don't have to do that much mental arithmetic and run into frustrating errors during setup.
+/// If we want to test trying to join with more than what we have, we can do so in the integration test.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Subnet {
+    /// The account we use to create the subnet.
+    pub creator: AccountId,
+    /// Collateral of the initial validator set.
+    ///
+    /// These validators will join the subnet with these collaterals after the subnet is created.
+    pub validators: CollateralMap,
+    /// Balances of the accounts at the creation of the subnet.
+    ///
+    /// These accounts will pre-fund the subnet after it's created.
+    pub balances: BalanceMap,
+    /// Nodes that participate in running the chain of this subnet.
+    pub nodes: NodeMap,
+    /// Relayers that submit bottom-up checkpoints to the parent subnet.
+    pub relayers: RelayerMap,
+    /// Child subnets under this parent.
+    ///
+    /// The subnet ID exists so we can find the outcome of existing deployments in the log.
+    pub subnets: SubnetMap,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Node {
+    /// Indicate whether this is a validator node or a full node.
+    pub mode: NodeMode,
+    /// Indicate whether to run the Ethereum API.
+    pub ethapi: bool,
+    /// The nodes from which CometBFT should bootstrap itself.
+    pub seed_nodes: Vec<NodeId>,
+    /// The parent node that the top-down syncer follows;
+    /// or leave it empty if the parent is CalibrationNet.
```

Comment:
Why do we sneakily assume Calibnet here when empty? How can we model this explicitly?

Reply:
Good question, it would be an external IP address of an Ethereum JSON-RPC endpoint, so everyone can specify their favourite parent node instance. AFAIK the glif.io or address that we tend to use in tests has load balancing, so it's the same for every node (which is wrong). 

We can add an enum to capture this.

Reply:
Added the `ParentNode` enum type.